x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlCCSUtils.java

```diff
@@ -304,24 +337,29 @@ static void updateExecutionInfoAtEndOfPlanning(EsqlExecutionInfo execInfo) {
     /**
      * Checks the index expression for the presence of remote clusters.
      * If found, it will ensure that the caller has a valid Enterprise (or Trial) license on the querying cluster
-     * as well as initialize corresponding cluster state in execution info.
+     * as well as initialize the corresponding cluster state in execution info.
      * @throws org.elasticsearch.ElasticsearchStatusException if the license is not valid (or present) for ES|QL CCS search.
      */
     public static void initCrossClusterState(
         IndicesExpressionGrouper indicesGrouper,
         XPackLicenseState licenseState,
-        IndexPattern indexPattern,
+        Set<IndexPattern> indexPatterns,
         EsqlExecutionInfo executionInfo
     ) throws ElasticsearchStatusException {
-        if (indexPattern == null) {
+        if (indexPatterns == null || indexPatterns.isEmpty()) {
             return;
         }
         try {
-            var groupedIndices = indicesGrouper.groupIndices(
-                IndicesOptions.DEFAULT,
-                Strings.splitStringByCommaToArray(indexPattern.indexPattern()),
-                false
-            );
+            String[] indexExpressions = indexPatterns.stream()
+                .map(indexPattern -> Strings.splitStringByCommaToArray(indexPattern.indexPattern()))
```

Comment:
This seems like it is combining all patterns into one. Unfortunately this is not safe to do.
For example:
* logs-*,-*2025
* metrics-*
If above 2 patterns are combined, metrics-2025 will be unintentionally excluded from the final result set.
Here this result is used to initialize CC state. If metrics-2025 is the only index on `r1` it will not be initialized. This would lead to runtime error when populating the cluster info when actually querying `r1`.

I am currently working to avoid the entire `initCrossClusterState`, so likely this will no longer be needed.
See https://github.com/elastic/elasticsearch/pull/136804

Reply:
OK. This problem seems to be a potential bug for subqueries and views in CCS, but should not cause any problems for the normal single FROM command case. We can merge this early, and fix the problem properly with #136804.